### PR TITLE
feat: add json streaming to JSONReader

### PR DESCRIPTION
**Pull request by KindOfAScam**
added streaming, refactored and cleaned up the JSON Reader class.

I did test with various JSON files ranging from a few hundred MBs up to a GB, but definitely could use some more testing to find edge cases. 

(cc: @ozgur142)

### CODE REVIEW

**Review by himself65 — APPROVED** [review]

File: packages/llamaindex/src/readers/JSONReader.ts [hunk 1]
```diff
@@ -1,71 +1,199 @@
+import { parseChunked } from "@discoveryjs/json-ext";
 import type { JSONValue } from "@llamaindex/core/global";
 import { Document, FileReader } from "@llamaindex/core/schema";
+import { ReadableStream } from "@llamaindex/env";
```
Comment (himself65) [inline packages/llamaindex/src/readers/JSONReader.ts:4]:
```suggestion
```
